### ISSUE TITLE
input bug in list display mode

### ISSUE THREAD

**Issue by SergeA** [sanskrit-lexicon/csl-apidev#34]
Hello!
Today we have a bug with input in List display mode.
If I input in HK and it is selected in preferences ( Input display - Same as keyboard ) it works ok.
But if it is selected ( Input display - Standard Romanization ) or ( Input display - Devanagari ) it uses wrong input scheme and does not find anything.

![image](https://user-images.githubusercontent.com/6435011/205125736-856aad59-afb1-483f-b737-942a40d14c56.png)

Also there is long time problem with English-Sanskrit dictionaries in List display mode.
Input is forced to Roman transliteration and does not allow to type English letters like w or z, which are converted to ṭ or ś.

**Comment by funderburkjim** [comment]
Hi, Serge.
  I do confirm HK/ problem
  For reference, here is what the 'Preferences'  looks like in the problem case:
![image](https://user-images.githubusercontent.com/6393033/205151144-67922bf5-9b4d-42d4-a363-3e46375dd966.png)

QUESTION:  Is this a NEW problem  (i.e.,  did the same set of preferences work properly until now? -

**Comment by funderburkjim** [comment]
Another user reported similar problem with List display.

My suspicion:  The server recently upgraded PHP to version 8.  I suspect the list display problem is somehow related to
this upgrade.

**Comment by funderburkjim** [comment]
The above two commits solve the List-display problem -- at least I think so!
@SergeA please exercise the list display some.

As to 'long time problem with English-Sanskrit dictionaries in List display mode', I don't find any noticeable delay in MWE searches.  Do you still have long delays here?

**Comment by SergeA** [comment]
> The above two commits solve the List-display problem 

Yes, it works again. Thank you very much!

> As to 'long time problem with English-Sanskrit dictionaries in List display mode', I don't find any noticeable delay in MWE searches. Do you still have long delays here?

No,no, I mean an input problem.
This problem I have in List display of English to Sanskrit dictionaries. The input works as transliteration. When I try to type e.g. "window" it is converted to "ṭindoṭ" etc. There is no way to search words with Q W F Z X letters.

But in Basic display the input is ok, and is written as ( input:  English, lower-case ) and it is not changeable.

**Comment by gasyoun** [comment]
> There is no way to search words with Q W F Z X letters.

You use the List mode for what reason? How is SIMPLE worse than List mode?

**Comment by funderburkjim** [comment]
## mwe and w, etc.

The list display DOES work for the English-Sanskrit dictionaries IF 
'Input display' is set to 'Same as keyboard input'.  Not sure if this
requirement should be considered a bug or a feature.


![image](https://user-images.githubusercontent.com/6393033/205346060-f41531e7-bab6-475f-baef-41b6745244ef.png)

![image](https://user-images.githubusercontent.com/6393033/205346329-97180efe-4693-49e4-b90f-0afcfecb1c17.png)

**Comment by SergeA** [comment]
> The list display DOES work for the English-Sanskrit dictionaries IF
> 'Input display' is set to 'Same as keyboard input'. Not sure if this
> requirement should be considered a bug or a feature.

Ah. It's true, it works. Now I know the way. )) Thank you, Jim!
Then for me it is ok. 
But it looks rather like a bug.

**Comment by SergeA** [comment]
> You use the List mode for what reason? How is SIMPLE worse than List mode?

There are certain reasons. Simple search is created for uneducated people and is very bad from the spelling training point of view. Sometimes it is unpredictably buggy and does not find the correct spelling. So I usually avoid using it, nor recommend to use it to people. And another reason is that I do not see Apte's English to Sanskrit dictionary in the Simple search menu. Then how can I use it there?

**Comment by gasyoun** [comment]
>  I do not see Apte's English to Sanskrit dictionary in the Simple search menu.

Yeah, this is what wrote about @funderburkjim some time ago. I believe it's fixable, switching off the hints for English words or having them in a different mode?

**Comment by gasyoun** [comment]
<details>
<summary>🤖 Bug triage findings</summary>

**Bug class:** display-rendering (frontend input widget behavior)

**Evidence:**

The issue contains two distinct sub-problems:

**Sub-problem 1 (HK input / Input display preference ignored in List mode):** Resolved. funderburkjim confirmed the fix in thread (2022-12-02) and SergeA confirmed it works. Code audit of `parm.php` line 145–150 confirms that for non-English dictionaries, `filterin` is correctly applied via `transcoder_processString($keyin1, $this->filterin, 'slp1')`.

**Sub-problem 2 (English-Sanskrit List mode forces transliteration; w/z converted to others):** Still open. `parm.php` line 145 shows `if ($this->english)` correctly bypasses transliteration on the server side — the bug is in the frontend input widget. The browser-side keystroke handler does not check the dictionary english flag before converting keystrokes. The local `js/listview.js` only handles list-click navigation; the keyboard input handler for the search box lives in display-layer JS not in this repo. Workaround: set Input display to Same as keyboard input.

**Agent-fixable (Tier A)?** No — needs display layer (frontend JS that intercepts keystrokes in List display input field).

**If Tier D — what needs checking:**
- The JS file handling keyboard input in List display mode (likely in csl-websanlexicon or sibling frontend repo)
- Input handler should check if dictinfo.english == true and skip transliteration when true, mirroring the server-side logic already in parm.php line 145

**Suggested next step:** Search csl-websanlexicon for the keyboard handler that applies transliteration in List display mode and add an English-dict guard.

</details>

**Comment by gasyoun** [comment]
H1523: English list-input half (AE/MWE/BOR window→ṭindoṭ) fixed in csl-websanlexicon PR https://github.com/sanskrit-lexicon/csl-websanlexicon/pull/89 — list view skips phonetic keyboard for those dicts. Other half of #34 (view-as mismatch) left open if still repros after deploy.
